android/daemon/src/main/java/org/mozilla/firefox/vpn/daemon/VPNService.kt

```diff
@@ -360,6 +384,13 @@ class VPNService : android.net.VpnService() {
         // asked boot vpn from the OS
         val prefs = Prefs.get(this)
         prefs.edit().putString("lastConf", json.toString()).apply()
+        
+        // Auto-enable start on boot when VPN connects successfully
```

Comment:
I'm not sure if this is desirable - "always on" exists in the Android VPN system settings and triggers a reboot, also it allows to block all network requests until the VPN is active which is more consistent.